File: app/src/androidTest/java/com/games/memorygame/injection/MemoryGameServiceMock.java
package com.games.memorygame.injection;

import com.games.memorygame.model.Photo;
import com.games.memorygame.model.Photos;
import com.games.memorygame.network.FlickrResponse;
import com.games.memorygame.network.MemoryGameService;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.Query;

public class MemoryGameServiceMock implements MemoryGameService {

    @Override
    public Observable<FlickrResponse> getPhotos(@Query("per_page") int perPage,
                                                @Query("text") String theme) {
        List<Photo> photos = new ArrayList<>();
        for (int i = 0; i < perPage; i++) {
            photos.add(new Photo("Title " + i, i, "", "", String.valueOf(i)));
        }
        return Observable.just(new FlickrResponse(new Photos(photos), "ok", null, ""));
    }

}

File: app/src/androidTest/java/com/games/memorygame/injection/MemoryGameServiceMock.kt
package com.games.memorygame.injection

import com.games.memorygame.model.Photo
import com.games.memorygame.model.Photos
import com.games.memorygame.network.FlickrResponse
import com.games.memorygame.network.MemoryGameService

import java.util.ArrayList

import io.reactivex.Observable
import retrofit2.http.Query

class MemoryGameServiceMock : MemoryGameService {

    override fun getPhotos(@Query("per_page") perPage: Int,
                           @Query("text") theme: String): Observable<FlickrResponse> {
        val photos = ArrayList<Photo>()
        (0 until perPage).forEach { photos.add(Photo("Title " + it, it, "", "", it.toString())) }
        return Observable.just(FlickrResponse(Photos(photos), "ok", null, ""))
    }

}
